### ISSUE TITLE
`no-literal-string` fails to detect strings when assigned to UPPER_SNAKE_CASE constants

### ISSUE THREAD

**Issue by shcshcshc** [edvardchen/eslint-plugin-i18next#148]
## Package version 
```
"@eslint/js": "^9.7.0",
"typescript-eslint": "^8.4.0",
"eslint": "^8.57.0",
"typescript-eslint": "^8.4.0",
"eslint-plugin-i18next": "^6.1.3",
```

## 🧪 Minimal Reproduction

With the following code:
```tsx
export const SNAKE_UPPER_CASE_EXAMPLE = "this should be error";
export const camelCaseExample = "thisShouldBeError";
```

And this rule configuration:
```tsx
"i18next/no-literal-string": [
  "error",
  {
    mode: "all",
  },
],
```

➡️ Expected: Both cases should be flagged as literal strings
❌ Actual: SNAKE_UPPER_CASE_EXAMPLE is not reported.

<img width="743" height="47" alt="Image" src="https://github.com/user-attachments/assets/b909f16e-b1a7-43b2-8533-02210cdfcde4" />

Is the behavior where objects assigned to SNAKE_UPPER_CASE constants are excluded from reporting intentional?
If so, I would like to know how I can report or handle this behavior.
If not, I believe this behavior is a bug, and that's why I'm reporting it.

**Comment by edvardchen** [comment]
This is by design.

The goal of this rule is to catch accidental literal strings, not every intentional string in the codebase.  assignments are treated as deliberate constants and are excluded on purpose.

Reporting those cases would add a lot of noise and drown out the cases this plugin is actually meant to catch.
